### PR TITLE
grt: get proper track spacing for tracks with multiple patterns

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/issues/1788

### CODE REVIEW

**Comment by eder-matheus** [comment]
@maliberty I've started a secure-ci run.

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -2675,27 +2675,10 @@ void GlobalRouter::computeSpacingsAndMinWidth(int max_layer)
...
-      min_width = track_step_y;
-    } else if (tech_layer->getDirection() == odb::dbTechLayerDir::VERTICAL) {
-      min_width = track_step_x;
-    }
+    int track_step, track_init, num_tracks;
+    track->getAverageTrackSpacing(track_step, track_init, num_tracks);
 
-    grid_->addSpacing(min_spacing, level - 1);
-    grid_->addMinWidth(min_width, level - 1);
+    grid_->addMinSpacing(track_step, level - 1);
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2681]: Is this a min spacing or a track pitch?  It seems like the second.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
Do you notice any effect on asap test cases?

**Comment by eder-matheus** [comment]
> Do you notice any effect on asap test cases?

I didn't look at the specific metrics yet, but nothing failed. Tomorrow I can look with more attention to the asap7 designs.

**Pull request merged**